exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -134,9 +140,13 @@ func (prwe *PrwExporter) PushMetrics(ctx context.Context, md pdata.Metrics) (int
 		if err := prwe.export(ctx, tsMap); err != nil {
 			dropped = pdatautil.MetricCount(md)
 			errs = append(errs, err)
+			permanent = permanent || consumererror.IsPermanent(err)
 		}
 
 		if dropped != 0 {
+			if permanent {
```

Comment:
Instead of doing this hack here, can you consider to change `componenterror.CombineErrors` to do this? you just need to check the type and if any error is permanent return permanent?

Probably a separate PR :)

Reply:
yeah. I was hesitant on doing that because i'm not sure if its good to introduce the idea of "conusmer" in component error. I have the code already haha. Will file a PR tomorrow then.

Reply:
Components already depend on consumer. The other way around is not expected

Reply:
This needs now to be removed, the `CombineErrors` handles this.